### ISSUE TITLE
Will replace text at tops of other pages with link title

### ISSUE THREAD

**Issue by Caffa** [zolrath/obsidian-auto-link-title#20]
Occurs when I move to another page when the link is being fetched. Has also occurred sometimes, when I click somewhere when the link is being fetched. 

Some text, at the top few lines of the page will be replaced with the title of the link and the link itself will remain "fetching title"

**Comment by eleanorkonik** [comment]
Came here to report this as well: I think I'm experiencing some mild data loss with the auto link title plugin. It seems like if I paste and the "back out" of the action, the text gets appended to the top of the file and weirdly deletes a chunk of whatever was already at the top of the file. I don't have time to do a deep dive confirming it,

**Comment by lizsugar** [comment]
I've also experienced this. I have to sit and wait for the plugin to finish pulling the title before I'm really able to do much else, else it destroys the top of my note

**Comment by lishid** [comment]
Upon inspection, it seems that this asynchronous operation doesn't check for whether the editor is from the same file here: https://github.com/zolrath/obsidian-auto-link-title/blob/0b2c5b804dbfc6e3b32bbef8d5d0d6f77adec969/main.ts#L100

@zolrath I suggest checking that the editor has the same file open with the same contents before applying the results, or at the very least, that you can find the position of the unique id.

**Comment by jamiekowalski** [comment]
@zolrath I implemented a [fix for this](https://github.com/jamiekowalski/obsidian-auto-link-title/commit/c837822de03679fa8c0afca5e3de426100d06b8e) that simply bails out if the hash isn't found. There are probably better approaches (e.g. if multiple editors open, be able to keep track of the one where the link title is loading; or if the file was closed, prompt the user to re-open it, etc.), but bailing out is at least better than overwriting the top of the file by using offset `-1`. If I get around to cleaning up the simple fix, I'll submit a pull request.
